Refactoring - Theme & FavouriteCities states

Lvlad27 opened the pull request:
- Theme and FavouriteCities states are now implemented through Redux and persisted in Local Storage
- Made some minor styling changes
- Remove unnecessary .env options

LvladCSol committed:
Initial commit

LvladCSol committed:
Init

LvladCSol committed:
change prettier indentation

LvladCSol committed:
update gitignore

LvladCSol committed:
add backend for api token data request

LvladCSol committed:
feat: serve token from api

LvladCSol committed:
add expiration time

LvladCSol committed:
adsa

LvladCSol committed:
Add token expiration time

LvladCSol committed:
update .gitignore

LvladCSol committed:
add geolocation function

LvladCSol committed:
testing the api

LvladCSol committed:
fix observations setResourceType call

LvladCSol committed:
add cors and dotenv packages

LvladCSol committed:
removed comment

LvladCSol committed:
chore: changed indentation

LvladCSol committed:
docs: change server script path

LvladCSol committed:
update geolocation function

LvladCSol committed:
add Inter fonts

LvladCSol committed:
add router and pages. Home Page main functionality

LvladCSol committed:
add initial css structure

LvladCSol committed:
add endpoints for api requests

LvladCSol committed:
remove comments

LvladCSol committed:
formatting

LvladCSol committed:
formatting

LvladCSol committed:
add some files

LvladCSol committed:
reinstall some packages

LvladCSol committed:
add extra variables for color & font size

LvladCSol committed:
moved to Home dir

LvladCSol committed:
add background color

LvladCSol committed:
add weather symbol icon endpoint

LvladCSol committed:
add axios config option to stringify response

LvladCSol committed:
adjust style for navbar component

LvladCSol committed:
add CurrWeatherCard comp. style & functionality

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into Vlad_Leontescu

LvladCSol committed:
comments cleaned

LvladCSol committed:
refactor: Loading and Error state

LvladCSol committed:
Merge branch 'Vlad_Leontescu' of github.com:Lvlad27/WeatherApp into HomePage

LvladCSol committed:
configure prettier indentation for 2 tabs

LvladCSol committed:
auto config indentation to 2 tabs

LvladCSol committed:
add JSON parsing to axios config

LvladCSol committed:
refactor

LvladCSol committed:
refactor: destructure props

LvladCSol committed:
minor change

LvladCSol committed:
docs: rename files, add symbol endpoint and function

LvladCSol committed:
add search bar component markup

LvladCSol committed:
add DailyWeather component

LvladCSol committed:
add weather details to main card component

LvladCSol committed:
change some styling

LvladCSol committed:
update for DailyWeather and Search children components

LvladCSol committed:
Merge remote-tracking branch 'upstream/Leontescu_Vlad' into Vlad_Leontescu

# Conflicts:
#	src/Components/CurrWeatherCard/CurrWeatherCard.jsx
#	src/Components/CurrWeatherCard/CurrWeatherCard.module.scss
#	src/Components/Navbar/Navbar.module.scss
#	src/Components/index.js
#	src/Helpers/endpoints.js
#	src/Helpers/geolocation.js
#	src/Pages/Home/Home.jsx
#	src/Pages/Home/Home.module.scss

LvladCSol committed:
refactor: formatDate is now a helper function

LvladCSol committed:
refactor: change props to currWeather.current

LvladCSol committed:
formatting update

LvladCSol committed:
search bar component

LvladCSol committed:
Merge branch 'Vlad_Leontescu' of github.com:Lvlad27/WeatherApp into Vlad_Leontescu

LvladCSol committed:
refactor: date formatting options

LvladCSol committed:
formatting

LvladCSol committed:
add search by location name endpoint

LvladCSol committed:
search bar autocomplete

LvladCSol committed:
Merge branch 'SearchBar' into Vlad_Leontescu

LvladCSol committed:
docs: folder and file restructure

LvladCSol committed:
add footer

LvladCSol committed:
change style and add main element

LvladCSol committed:
add hourly forecast component

LvladCSol committed:
current weather now changes based on search input

LvladCSol committed:
change header color

LvladCSol committed:
remove header styles comments

LvladCSol committed:
remove console.log

LvladCSol committed:
changes to webpack

LvladCSol committed:
restructure dataService files

LvladCSol committed:
Add Feedback Form component

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into FeedbackForm

LvladCSol committed:
Merge branch 'master' of github.com:Lvlad27/WeatherApp

LvladCSol committed:
fetched upstream

LvladCSol committed:
remove isSubmit state
move cross icon to Error Message Components
replace br tag with css

LvladCSol committed:
Merge branch 'FeedbackForm'

LvladCSol committed:
install react-icons

LvladCSol committed:
Add Favorite Cities Component

LvladCSol committed:
minor styling change

LvladCSol committed:
fix loadState and saveState

LvladCSol committed:
update FavPage

LvladCSol committed:
Merge branch 'FavoriteCities'

LvladCSol committed:
add themeContext

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into ThemeSwitch

Lvlad27 committed:
Update ThemeSwitch.jsx

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask

LvladCSol committed:
line format

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask

LvladCSol committed:
Merge branch 'master' into ThemeSwitch

LvladCSol committed:
Merge branch 'ThemeSwitch' of github.com:Lvlad27/WeatherApp into ThemeSwitch

LvladCSol committed:
resolved conflict

LvladCSol committed:
Dark/Light theme styling and scss refactoring.

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask into AboutPage

LvladCSol committed:
create About Page

LvladCSol committed:
add more images

LvladCSol committed:
More improvements

Add reusable Button component and refactor components accordingly.
Minor refactoring.

LvladCSol committed:
refactoring

remove save to local storage function
change ErrorMessage folder
change name of saved locations

LvladCSol committed:
refactor Redux for Theme and FavLocations

LvladCSol committed:
Combined reducers refactoring

- Theme and Favourite Cities states are now implemented through Redux and persisted in Local Storage
- Made some minor styling changes
- Remove unnecessary .env options

LvladCSol committed:
remove unnecessary imports

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into Refactor

JuliaShavyrova commented on src/Components/Header/ThemeSwitch/ThemeSwitch.jsx:
```diff
@@ -1,16 +1,32 @@
+              defaultChecked={false}
+              id="checkbox-toggle"
+              onClick={() => dispatch(toggleTheme())}
+            />
```
why not like this? or just make this input controlled
```
<input
  type="checkbox"
  checked={theme === "light"}
  id="checkbox-toggle"
  onClick={() => dispatch(toggleTheme())}
/>
```

Lvlad27 replied:
ok, fixed

JuliaShavyrova commented on src/redux/index.js:
```diff
@@ -1,48 +1,24 @@
-  saveState(store.getState());
+  saveState('savedLocations', store.getState().favCityList);
+  saveState('theme', store.getState().theme);
+  console.log(store.getState());
```
please clean code

Lvlad27 replied:
done

JuliaShavyrova commented on src/context/themeContext.js:
```diff
@@ -1,17 +1,16 @@
     this.setState((prevState) => {
       return {
-        theme: prevState.theme === 'light' ? 'dark' : 'light'
+        theme: prevState.theme === 'dark' ? 'light' : 'dark'
```
do you still need theme provider after you created theme reducer?

Lvlad27 replied:
fixed

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask into Refactor

LvladCSol committed:
Merge branch 'Refactor' of github.com:Lvlad27/WeatherApp into Refactor

LvladCSol committed:
reformat

LvladCSol committed:
PR COMMENTS + custom hook

- remove console.log and themeContext from all Components
- add custom hook for all async requests

LvladCSol committed:
refactor themeSwitch input

JuliaShavyrova commented on src/hooks/useApi.js:
```diff
@@ -0,0 +1,103 @@
+    getCurrentWeatherByCoords();
+    getLocationDataByCoords();
+    getDailyWeather();
+    getHourlyWeather();
```
if you want to get all responses at once, you can try axios.all:
```
axios.all([
  instance.get(endpoints.LOCATION_DATA_BY_COORDS(coords)), 
  instance.get(endpoints.HOURLY_WEATHER(coords)), 
  ....
])
```

Lvlad27 replied:
done

LvladCSol committed:
add missing dispatch assignment line

LvladCSol committed:
refactor useApi for concurrent requests

JuliaShavyrova commented on src/hooks/useApi.js:
```diff
@@ -0,0 +1,64 @@
+        })
+      )
+      .catch((error) => {
+        setError(error);
```
you need to set loading=false here if you don't want to get infinite loading status :)

Lvlad27 replied:
Yes, actually I put it right after the commit. Fixed now

LvladCSol committed:
refactor

- logo is now clickable and goes to homepage
- Message box now added for error and loading states
- set loading state when catching error
